proto/tendermint/abci/types.proto

```diff
@@ -117,6 +118,10 @@ message RequestApplySnapshotChunk {
   string sender = 3;
 }
 
+message RequestPrepareProposal {
+  repeated bytes block_data = 1;
```

Comment:
@ValarDragon what you think of sending the bytes to the app as well so it knows how much space it has to work with, avoid extra calculation by the app?

Reply:
What is `block_data` in this context and how will the app use it? Also, can we add a godoc to this type please?

Reply:
block_data here are the transactions. The application could aggregate the signatures, include a proof of certain txs. These are only two use cases that come to mind, currently. 

yes, will get the godoc in

Reply:
Sorry didn't see the ping to me.

So with #6710, prepare proposal will take in the header as well. So is there any data thats not included in the proposal as well?

Reply:
@ValarDragon If the application would like to house the mempool, it must know the amount of txs it can add. 

For this reason, I think it would be good to add the amount of bytes that can fit in the block to prepareProposal.